src/core/transport/chttp2/stream_lists.c

```diff
@@ -313,12 +313,26 @@ int grpc_chttp2_list_pop_check_read_ops(
   return r;
 }
 
-void grpc_chttp2_list_add_stalled_by_transport(
+void grpc_chttp2_list_add_writing_stalled_by_transport(
     grpc_chttp2_transport_writing *transport_writing,
     grpc_chttp2_stream_writing *stream_writing) {
   stream_list_add(TRANSPORT_FROM_WRITING(transport_writing),
                   STREAM_FROM_WRITING(stream_writing),
-                  GRPC_CHTTP2_LIST_STALLED_BY_TRANSPORT);
+                  GRPC_CHTTP2_LIST_WRITING_STALLED_BY_TRANSPORT);
+}
+
+void grpc_chttp2_list_flush_writing_stalled_by_transport(
+    grpc_chttp2_transport_writing *transport_writing) {
```

Comment:
Suggest:

```c
void grpc_chttp2_list_flush_writing_stalled_by_transport(grpc_chttp2_transport_writing *transport_writing, bool is_window_available);
```

and move the `transport_writing->outgoing_window > 0` check into writing.c, to avoid flow control logic creeping into the list management code.